app/src/main/java/appricottsoftware/clarity/HomeActivity.java

```diff
@@ -568,4 +574,49 @@ public void onComplete(@NonNull Task<Void> task) {
         }
         clarityApp.clearGoogleSignInClient();
     }
+
+    // This method interfaced within ChannelFragment
+    @Override
+    public void sendChannels(List<Channel> channels) {
+        // Get reference to BrowseFragment
+        String homeFragmentTag = getString(R.string.home_fragment_tag);
+        Fragment page = getSupportFragmentManager().findFragmentByTag(homeFragmentTag);
+        HomeFragment homeFragment = (HomeFragment) page;
+
+        // If successful use HomeFragment as medium to communicate between Channels and Browse
+        if (homeFragment != null) {
+            Log.i(TAG, "Receive callback from ChannelFragment, sending data to browse");
+            homeFragment.sendDataToBrowseFragment(channels);
+        }
```

Comment:
What happens if homeFragment is null? This might not occur at all, but we should handle it. I'm okay with a simple Log.e, but maybe we can throw something useful. 

If it's never null, get rid of the if statement

Reply:
I think the app wouldn't load much if homeFragment is null. HomeFragment currently holds the 2 tabs at the top which encapsulates Channel and Browse Fragments. I'll throw a Log.e in the meantime